crates/mpc-attestation/build.rs

```diff
@@ -0,0 +1,187 @@
+use std::env;
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::Path;
+use std::path::PathBuf;
+
+const ASSETS_DIR_NAME: &str = "assets";
+
+fn main() {
+    // Find assets directory
+    let manifest_dir =
+        env::var("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR must be set by Cargo");
+
+    let assets_dir = PathBuf::from(&manifest_dir).join(ASSETS_DIR_NAME);
+
+    println!("cargo:rerun-if-changed={}", ASSETS_DIR_NAME);
+
+    // First pass: ensure directory exists + register rerun triggers
+    let entries = fs::read_dir(&assets_dir).unwrap_or_else(|e| {
+        panic!(
+            "Failed to read assets directory '{}': {}.\n\
+             This directory must exist and contain tcb_info*.json files.",
+            assets_dir.display(),
+            e
+        )
+    });
+
+    let mut measurement_files = Vec::new();
+
+    for entry in entries {
+        let entry = entry.unwrap_or_else(|e| {
+            panic!(
+                "Failed to read an entry inside '{}': {}",
+                assets_dir.display(),
+                e
+            )
+        });
+
+        let path = entry.path();
+
+        if path.extension().and_then(|x| x.to_str()) == Some("json") {
+            println!("cargo:rerun-if-changed={}", path.display());
```

Comment:
I don’t think we need it since we already have it for the parent directory. See: https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-changed:
> If the path points to a directory, it will scan the entire directory for any modifications.

But please test it to double-check.